metadata-jobs/mae-consumer/src/test/java/com/linkedin/metadata/kafka/hook/event/PlatformEventMatcher.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.metadata.kafka.hook.event;
+
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.PlatformEvent;
+import com.linkedin.platform.event.v1.EntityChangeEvent;
+import javax.annotation.Nonnull;
+import org.mockito.ArgumentMatcher;
+
+
+public class PlatformEventMatcher implements ArgumentMatcher<PlatformEvent> {
+  private final PlatformEvent _expected;
+
+  public PlatformEventMatcher(@Nonnull final PlatformEvent expected) {
+    _expected = expected;
+  }
+
+  @Override
+  public boolean matches(@Nonnull final PlatformEvent actual) {
+    return _expected.getName().equals(actual.getName()) && _expected.getHeader()
+        .getTimestampMillis()
+        .equals(actual.getHeader().getTimestampMillis()) && payloadMatches(actual);
+  }
+
+  public boolean payloadMatches(@Nonnull final PlatformEvent actual) {
```

Comment:
This is really cool. I've never used this but going to now

Reply:
Haha thanks, yeah it was very clutch